packages/config/src/normalize.ts

```diff
@@ -3,33 +3,37 @@ import { mergeConfigs } from './merge.js'
 import { DEFAULT_ORIGIN } from './origin.js'
 import { removeFalsy } from './utils/remove_falsy.js'
 
-// Normalize configuration object
-export const normalizeConfig = function (config) {
+/**
+ * Normalize configuration object
+ */
+export const normalizeConfig = function (config: $TSFixMe, packagePath?: string) {
   const configA = removeEmpty(config)
-  const { build, functions, plugins, ...configB } = mergeConfigs([DEFAULT_CONFIG, configA])
+  const { build, functions, plugins, ...configB } = mergeConfigs([DEFAULT_CONFIG(packagePath), configA]) as $TSFixMe
   const { build: buildA, functions: functionsA, functionsDirectoryProps } = normalizeFunctionsProps(build, functions)
   const pluginsA = plugins.map(normalizePlugin)
   return { ...configB, build: buildA, functions: functionsA, plugins: pluginsA, ...functionsDirectoryProps }
 }
 
-// Remove empty strings.
-// This notably ensures that empty strings in the build command are removed.
-// Otherwise, those would be run during builds, making the build fail.
+/**
+ * Remove empty strings.
+ * This notably ensures that empty strings in the build command are removed.
+ * Otherwise, those would be run during builds, making the build fail.
+ */
 const removeEmpty = function ({ build, ...config }) {
   return removeFalsy({ ...config, build: removeFalsy(build) })
 }
 
-const DEFAULT_CONFIG = {
+const DEFAULT_CONFIG = (packagePath?: string) => ({
   build: {
     environment: {},
-    publish: '.',
+    publish: packagePath || '.',
```

Comment:
This is for `Option A` inside the Slack discussion: https://netlify.slack.com/archives/C05556LEX28/p1691423437855069